metricflow/dataflow/dataflow_plan.py

```diff
@@ -178,22 +178,15 @@ def visit_join_conversion_events_node(self, node: JoinConversionEventsNode) -> V
 class DataflowPlan(MetricFlowDag[DataflowPlanNode]):
     """Describes the flow of metric data as it goes from source nodes to sink nodes in the graph."""
 
-    def __init__(  # noqa: D107
-        self, sink_output_nodes: Sequence[DataflowPlanNode], plan_id: Optional[DagId] = None
-    ) -> None:
-        if len(sink_output_nodes) == 0:
-            raise RuntimeError("Can't create a dataflow plan without sink node(s).")
-        self._sink_output_nodes = tuple(sink_output_nodes)
+    def __init__(self, sink_nodes: Sequence[DataflowPlanNode], plan_id: Optional[DagId] = None) -> None:  # noqa: D107
+        assert len(sink_nodes) == 1, f"Exactly 1 sink node is supported. Got: {sink_nodes}"
```

Comment:
Out of curiosity, are we going to formalize this via the type system?

Reply:
I haven't had a chance to think about how to simplify this, but yeah, that sounds like a good idea.